app/models/git_repository.rb

```diff
@@ -8,6 +8,7 @@ class GitRepository < ApplicationRecord
 
   has_many :git_branches, :dependent => :destroy
   has_many :git_tags, :dependent => :destroy
+  before_destroy :delete_repo_dir
```

Comment:
@mkanoor How big is the exposure here?  Is syncing the actions something required for this PR or would that be separate?

Also, I think the concern @Fryguy has is a little different then multiple tasks running.  If we fail when removing the files from disk it would leave them in an incomplete state, but the `GitRepository` instance would be intact.